src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -57,133 +56,79 @@ public BlobFeedAction(string expectedFeedUrl, string accountKey, string indexDir
+                };
 
-                    await UploadIndexJson(clientThrottle, true, packageDirectory, listAzureBlobs);
-                }
-            }
-            catch (Exception e)
-            {
-                Log.LogErrorFromException(e);
-            }
-
-            return !Log.HasLoggedErrors;
-        }
-
-        private async Task UploadIndexJson(SemaphoreSlim clientThrottle, bool allowOverwrite, string packageDirectory, List<string> listAzureBlobs)
-        {
-            listAzureBlobs.Remove(listAzureBlobs.Find(x => x.Contains("index.json")));
-            List<string> updatedVersions = new List<string>();
-            foreach (var version in listAzureBlobs)
-            {
-                string versionToCheck = version.Substring(packageDirectory.Length + 1).Split('/')[0];
-                NuGetVersion nugetVersion = null;
-                if (NuGetVersion.TryParse(versionToCheck, out nugetVersion))
+                LocalSettings settings = new LocalSettings
                 {
-                    updatedVersions.Add(versionToCheck);
-                }
-            }
-            string packageIndexJsonLocation = feed.GeneratePackageServiceIndex(packageDirectory, updatedVersions);
-            await UploadAsync(CancellationToken, packageIndexJsonLocation, packageDirectory + "/index.json", clientThrottle, true);
-        }
-
-        private async Task UploadAsync(CancellationToken ct, string item, string uploadPath, SemaphoreSlim clientThrottle, bool allowOverwrite)
-        {
-            if (!File.Exists(item))
-                throw new Exception(string.Format("The file '{0}' does not exist.", item));
+                    Json = JObject.FromObject(sleetSettings)
+                };
 
-            await clientThrottle.WaitAsync();
-            string leaseId = string.Empty;
-            //this defines the lease for 15 seconds (max is 60) and 3000 milliseconds between requests
-            AzureBlobLease blobLease = new AzureBlobLease(feed.AccountName, feed.AccountKey, string.Empty, feed.ContainerName, uploadPath, Log, "15", "3000");
+                CloudStorageAccount storageAccount = CloudStorageAccount.Parse(source.ConnectionString);
+                AzureFileSystem fileSystem = new AzureFileSystem(new LocalCache(), new Uri(source.Path), new Uri(source.Path), storageAccount, source.Name, source.FeedSubPath);
 
-            bool isLeaseRequired = allowOverwrite && await feed.CheckIfBlobExists(uploadPath);
-
-            if (isLeaseRequired)
-            {
-                try
-                {
-                    leaseId = blobLease.Acquire();
-                }
-                catch (Exception)
+                // In case the first Push attempt fails with an InvalidOperationException we Init the feed and retry the Push command once.
+                // Sleet internally retries 5 times on each package when the push operation fails so we don't need to retry ourselves.
+                for (int i = 0; i <= 1; i++)
```

Comment:
Sleet does the retries. Is this loop needed?

Reply:
Yes in case an Init is needed, this loop does not retry on op failure

Reply:
But it only loops once?

Reply:
Think of it as a do/while without the need of having extra variables in place. If the call works then we just return if it fails with an expected exception it tried to init just once, else it fails

Reply:
It seems like it would be clearer/simpler to make the calls more self-contained and write out the logic.

```C#
try
{
    return await push();
}
catch (InvalidOperationException ex) when (ex.Message.Contains("init"))
{
    if (await init())
    {
        return await push();
    }
    return false;
}
```

(I'm imagining using async funcs as a minimal-change way of getting it done.)